src/handlers/swap_quote_handlers.ts

```diff
@@ -0,0 +1,199 @@
+    }
+    public async calculateSwapQuoteAsync(
+        buyTokenAddress: string,
+        sellTokenAddress: string,
+        buyAmount: BigNumber | undefined,
+        sellAmount: BigNumber | undefined,
+        from: string | undefined,
+        isETHSell: boolean = false,
+        slippagePercentage?: number,
+        providedGasPrice?: BigNumber,
+    ): Promise<GetSwapQuoteResponse> {
+        let swapQuote;
+        if (sellAmount !== undefined) {
+            swapQuote = await this._swapQuoter.getMarketSellSwapQuoteAsync(
+                buyTokenAddress,
+                sellTokenAddress,
+                sellAmount,
+                { slippagePercentage, gasPrice: providedGasPrice },
+            );
+        } else if (buyAmount !== undefined) {
+            swapQuote = await this._swapQuoter.getMarketBuySwapQuoteAsync(
+                buyTokenAddress,
+                sellTokenAddress,
+                buyAmount,
+                { slippagePercentage, gasPrice: providedGasPrice },
+            );
+        } else {
+            throw new Error('sellAmount or buyAmount required');
+        }
+        const {
+            makerAssetAmount,
+            totalTakerAssetAmount,
+            protocolFeeInWeiAmount: protocolFee,
+        } = swapQuote.worstCaseQuoteInfo;
+        const { orders, gasPrice } = swapQuote;
+
+        // If ETH was specified as the token to sell then we use the Forwarder
+        const extensionContractType = isETHSell ? ExtensionContractType.Forwarder : ExtensionContractType.None;
+        const {
+            calldataHexString: data,
+            ethAmount: value,
+            toAddress: to,
+        } = await this._swapQuoteConsumer.getCalldataOrThrowAsync(swapQuote, {
+            useExtensionContract: extensionContractType,
+        });
+
+        let gas;
+        if (from) {
+            gas = await this._estimateGasOrThrowRevertErrorAsync({
+                to,
+                data,
+                from,
+                value,
+                gasPrice,
+            });
+        }
+
+        const buyTokenDecimals = findTokenDecimalsIfExists(buyTokenAddress, CHAIN_ID) || DEFAULT_TOKEN_DECIMALS;
+        const sellTokenDecimals = findTokenDecimalsIfExists(sellTokenAddress, CHAIN_ID) || DEFAULT_TOKEN_DECIMALS;
+        const price = Web3Wrapper.toUnitAmount(makerAssetAmount, buyTokenDecimals)
+            .dividedBy(Web3Wrapper.toUnitAmount(totalTakerAssetAmount, sellTokenDecimals))
+            .decimalPlaces(sellTokenDecimals);
+
+        const apiSwapQuote: GetSwapQuoteResponse = {
+            price,
+            to,
+            data,
+            value,
+            gas,
+            from,
+            gasPrice,
+            protocolFee,
+            makerAssetAmount,
+            totalTakerAssetAmount,
+            orders,
+        };
+        return apiSwapQuote;
+    }
+    public async getSwapQuoteAsync(req: express.Request, res: express.Response): Promise<void> {
```

Comment:
nit: the soft convention I've been following is to keep these "handler" files only have interfaces that deal with `Request` and `Response`... then you put the rest of the logic in a "service" and have the service be a dependency of the handlers.